rhino/src/test/java/org/mozilla/javascript/tests/NativeRegExpTest.java

```diff
@@ -948,5 +948,14 @@ public void unicodeEscapeFallback() {
         final String script2 =
                 "var regex = /\\x/;\n" + "var res = '' + regex.test('x');\n" + "res;";
         Utils.assertWithAllModes_ES6("true", script2);
+
+        // rhino permits hex and unicode escapes with less than 2 and 4 digits respectively
+        final String script3 =
+                "var regex = /\\x1/;\n" + "var res = '' + regex.test('\\x01');\n" + "res;";
+        Utils.assertWithAllModes_ES6("true", script3);
+
+        final String script4 =
+                "var regex = /\\u61/;\n" + "var res = '' + regex.test('a');\n" + "res;";
+        Utils.assertWithAllModes_ES6("true", script4);
```

Comment:
Maybe splitting this test into four with descriptive names is an possible improvement. 
Personally i think debugging test cases with only one test inside is more effective. 
And having more test also improves the stats...

Reply:
Yeah, I agree that debugging is easier with one assertion in a test case. However it feels somewhat of an overkill in this case. I think I can combine them into one assertion, which seems to have best of both worlds :)